azurerm/internal/services/network/network_connection_monitor_resource.go

```diff
@@ -200,10 +200,30 @@ func resourceNetworkConnectionMonitor() *schema.Resource {
 							},
 						},
 
+						"resource_id": {
+							Type:     schema.TypeString,
+							Optional: true,
+							Computed: true,
+							ValidateFunc: validation.Any(
+								computeValidate.VirtualMachineID,
+								logAnalyticsValidate.LogAnalyticsWorkspaceID,
+							),
+						},
+
+						"type": {
+							Type:     schema.TypeString,
+							Optional: true,
+							ValidateFunc: validation.StringInSlice([]string{
+								string(network.MMAWorkspaceMachine),
+							}, false),
+						},
+
 						"virtual_machine_id": {
 							Type:         schema.TypeString,
 							Optional:     true,
+							Computed:     true,
 							ValidateFunc: computeValidate.VirtualMachineID,
+							Deprecated:   "Deprecated in favour of `resource_id` since the service API supports more resource types",
```

Comment:
```suggestion
							Type:          schema.TypeString,
							Optional:      true,
							Computed:      true,
							ConflictsWith: []string{"resource_id"},
							ValidateFunc:  computeValidate.VirtualMachineID,
							Deprecated:    "This property has been renamed to `resource_id` and will be removed in v3.0 of the provider.",

```

Reply:
After tested, ConflictsWith doesn't work in TypeSet and it would throw below error message. So I have to add logic to judge it.

```
Error: Internal validation of the provider failed! This is always a bug
with the provider itself, and not a user issue. Please report
this bug:

1 error occurred:
        * resource azurerm_network_connection_monitor: ConflictsWith: resource_id references unknown attribute (virtual_machine_id) at part (virtual_machine_id)
```